src/Core/Content/Category/DataAbstractionLayer/CategoryIndexer.php

```diff
@@ -115,21 +116,31 @@ public function update(EntityWrittenContainerEvent $event): ?EntityIndexingMessa
             );
         }
 
-        $children = $this->fetchChildren($ids, $event->getContext()->getVersionId());
-        $ids = array_unique(array_merge($ids, $children));
+        $onlyParentIds = array_diff($parentIds, $ids);
```

Comment:
you should add `array_unique` here, because the changed categories could have the same parent